src/stack.py

```diff
@@ -8,28 +8,19 @@ class Stack(object):
 
     def __init__(self, value_list=None):
         """Create an instance of a stack."""
-        if value_list:
-            self._linked_list = LinkedList(value_list)
-            self.is_empty = False
-        else:
-            self._linked_list = LinkedList()
-            self.is_empty = True
+        self._linked_list = LinkedList(value_list)
         self.top = self._linked_list.header
```

Comment:
Now that you've gotten this far, do you really need the `top` attribute? It doesn't look like you're actually referencing it anywhere for any purpose, only setting its value.